pyincore/models/hazard/tornado.py

```diff
@@ -0,0 +1,177 @@
+        """ Retrieve bulk earthquake hazard values either from the Hazard service or read it from local Dataset
+
+        Args:
+            payload (list):
+            hazard_service (obj): Hazard service.
+            kwargs (dict): Keyword arguments.
+        Returns:
+            obj: Hazard values.
+
+        """
+        if self.id and self.id != "" and hazard_service is not None:
+            return hazard_service.post_tornado_hazard_values(self.id, payload, **kwargs)
+        else:
+            if self.tornado_type == "dataset":
+                return self.calculate_wind_speed_uniform_random_dist(payload)
+            else:
+                raise ValueError("Local Tornado type \"" + self.tornado_type + "\" is not supported yet.")
+
+    def calculate_wind_speed_uniform_random_dist(self, payload):
+        """ Read local hazard values from shapefile dataset
+
+                Args:
+                    payload (list):
+                Returns:
+                    obj: Hazard values.
+
+                """
+
+        response = []
+
+        # match demand types with shapefile file
+        for req in payload:
+            hazard_values = []
+            for index, req_demand_type in enumerate(req["demands"]):
+                match = False
+                for hazard_dataset in self.hazardDatasets:
+                    if hazard_dataset.dataset is None or not isinstance(hazard_dataset.dataset, Dataset):
+                        raise Exception("Hazard dataset is not properly attached to the hazard object.")
+
+                    # find matching raster file (Dataset) to read value from
+                    if req_demand_type.lower() == hazard_dataset.demand_type.lower():
+                        hazard_df = hazard_dataset.dataset.get_dataframe_from_shapefile()
+                        ef_box = -1
+                        x = float(req["loc"].split(",")[0])
+                        y = float(req["loc"].split(",")[1])
+                        location = Point(x, y)
+
+                        for _, feature in hazard_df.iterrows():
+                            polygon = feature['geometry']
+                            if location.within(polygon):
+                                ef_rating = feature[self.EF_RATING_FIELD]
+                                ef_box = Tornado.get_ef_rating(ef_rating)
+                                break
+
+                        if ef_box < 0:
+                            return None
+
+                        if ef_box == 5:  # EF5
+                            bottom_speed = self.EF_WIND_SPEED[ef_box]
+                            top_speed = self.MAX_WIND_SPEED
+                        else:
+                            bottom_speed = self.EF_WIND_SPEED[ef_box]
+                            top_speed = self.EF_WIND_SPEED[ef_box + 1]
+
+                        # generate random wind speed
+                        random.seed(self.SEED)
```

Comment:
I think we need to match the service implementation here. A while back we decided to use the seed and get a random seed value based on what is set by the user. The default is -1 for seed, which indicates no seed is set and this method would handle that:

https://github.com/IN-CORE/incore-services/blob/develop/server/hazard-service/src/main/java/edu/illinois/ncsa/incore/service/hazard/models/tornado/utils/TornadoCalc.java#L73

If -1, then it will use the system time and randomize that. If a user sets a seed value when creating the dataset, it will use that.